### PR TITLE
enh(widgets): add Meta Services in Graph widget

### PR DESCRIPTION
**Pull request by lpinsivy**
## Description

Added possibility to display meta services in graph widget

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [x] 23.04.x (master)

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Comment by abuathier** [comment]
After reading the code + the ticket, to me, this PR need further discussions :
- the modified code seems "hacky" (2 decoupled `if` in 2 files for coupled concept)
- perfomance consideration with the added `UNION`

File: centreon/www/class/centreonWidget/Params/Connector/Service.class.php [hunk 1]
```diff
@@ -103,7 +103,9 @@ protected function getServiceIds($hostId)
         $res = $this->db->query($sql);
         $tab = array();
         while ($row = $res->fetchRow()) {
-            $tab[$hostId . "-" . $row['service_id']] = $row['service_description'];
+            # For meta services, use display_name column instead of service_description
+            $serviceDescription = (preg_match('/meta_/', $row['service_description'])) ? $row['display_name'] : $row['service_description'];
```
Comment (abuathier) [inline centreon/www/class/centreonWidget/Params/Connector/Service.class.php:107]: New code should respect the coding style ( < 120 characters on 1 line)

File: centreon/www/include/home/customViews/triggers/loadServiceFromHost.php [hunk 2]
```diff
@@ -87,7 +87,9 @@
         while ($row = $stmt->fetch(\PDO::FETCH_ASSOC)) {
             $xml->startElement('option');
             $xml->writeElement('id', $row['service_id']);
-            $xml->writeElement('label', $row['service_description']);
+            # For meta services, use display_name column instead of service_description
+            $serviceDescription = (preg_match('/meta_/', $row['service_description'])) ? $row['display_name'] : $row['service_description'];
```
Comment (abuathier) [inline centreon/www/include/home/customViews/triggers/loadServiceFromHost.php:91]: New code should respect the coding style ( < 120 characters on 1 line)

File: centreon/www/class/centreonWidget/Params/Connector/Service.class.php [hunk 3]
```diff
@@ -103,7 +103,9 @@ protected function getServiceIds($hostId)
         $res = $this->db->query($sql);
         $tab = array();
         while ($row = $res->fetchRow()) {
-            $tab[$hostId . "-" . $row['service_id']] = $row['service_description'];
+            # For meta services, use display_name column instead of service_description
```
Comment (abuathier) [inline centreon/www/class/centreonWidget/Params/Connector/Service.class.php:106]: The `#` character is unconventional for comments in 2023, except for specific Attribute case.
Please use `//`

File: centreon/www/include/home/customViews/triggers/loadServiceFromHost.php [hunk 4]
```diff
@@ -87,7 +87,9 @@
         while ($row = $stmt->fetch(\PDO::FETCH_ASSOC)) {
             $xml->startElement('option');
             $xml->writeElement('id', $row['service_id']);
-            $xml->writeElement('label', $row['service_description']);
+            # For meta services, use display_name column instead of service_description
```
Comment (abuathier) [inline centreon/www/include/home/customViews/triggers/loadServiceFromHost.php:90]: The `#` character is unconventional for comments in 2023, except for specific Attribute case.
Please use `//`

**Review by abuathier — APPROVED** [review]

**Review by wtermellil — APPROVED** [review]

**Pull request merged**